iib/web/errors.py

```diff
@@ -32,3 +34,45 @@ def json_error(error):
         response = jsonify({'error': msg})
         response.status_code = status_code
     return response
+
+
+def handle_broker_error(request_id, request_model):
+    """
+    Handle RabbitMQ error by sating request as failed and raises IIBError.
+
+    :param int request_id: Database ID of an failed request
+    :param Request request_model: type of an index image build request
+    """
+    req = db.session.query(request_model).get(request_id)
+
+    req.add_state("failed", "The scheduling of the request failed")
+    db.session.commit()
+    messaging.send_message_for_state_change(req)
+
+    current_app.logger.exception(f"The scheduling of the build request with ID {req.id} failed")
```

Comment:
Please assign this string to a variable so you don't have to duplicate it and do the string substitution twice.